### PR TITLE
Add a prop called component so as to pass a component for statusbar 

### PR DESCRIPTION
**Pull request by karkipy**
Add a prop called component so as to pass a component for statusbar and add a provision to use the props set for statusbar and change font color based on light prop, @syaau

### CODE REVIEW

**Comment by karkipy** [comment]
@syaau , using component as prop makes code a bit clean,
screen shot 2019-01-30 at 1 30 54 pm

**Comment by syaau** [comment]
```jsx
Inbox = () => (
   <StatusBar backgroundColor="#9D0001" light>
      <Search />
      <Title title="Inbox" />
      <Button onPress={this.onNew} icon={newIcon} />
   </StatusBar>
);
```
Which one looks cleaner ?

**Comment by syaau** [comment]
Besides, it also makes it extra clumsy to pass the parameters, like in your example, you would have to create another functional component just to pass the `title` prop.

File: StatusBar/StatusBar.js [hunk 1]
```diff
@@ -10,33 +10,35 @@ const styles = StyleSheet.create({
...
+    alignSelf: 'center',
   },
 });
 
 type Props = {
   title: string,
   backgroundColor: ?string,
   light?: boolean,
-  fontColor: ?string,
+  children: any,
```
Comment (syaau) [inline StatusBar/StatusBar.js]: Do not use any. It should either be a Node or an array of node.

File: StatusBar/StatusBar.js [hunk 2]
```diff
@@ -8,9 +8,14 @@ const barHeight = Platform.OS === 'ios' ? NativeModules.Screen.topMargin : 0;
...
     width: '100%',
-    minHeight: barHeight + 42,
+  },
+  body: {
+    marginTop: barHeight,
+    minHeight: 42,
+    width: '100%',
+    flexDirection: 'row',
+    alignItems: 'center',
     justifyContent: 'center',
```
Comment (syaau) [inline StatusBar/StatusBar.js]: space-inbetween

File: StatusBar/StatusBar.js [hunk 3]
```diff
@@ -40,13 +45,15 @@ class StatusBar extends Component<Props> {
...
         />
-        <Text
-          allowFontScaling={false}
-          style={[styles.title, { color: light ? 'white' : 'black' }]}
-        >
-          {title}
-        </Text>
-        {children}
+        <View style={styles.body}>
+          <Text
```
Comment (syaau) [inline StatusBar/StatusBar.js]: {title && <....>}

Comment (karkipy) [reply StatusBar/StatusBar.js]: we will have to wrap it in another view as space-inbetween applies for this as well, View with center will do it, is there other better solution?

**Comment by syaau** [comment]
@karkipy You should use the following style:
```javascript
titleText: {
    width: '100%',
    position: 'absolute',
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
  },
```
And get rid of that extra view.

**Pull request merged**